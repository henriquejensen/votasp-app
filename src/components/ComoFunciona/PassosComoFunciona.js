import React from 'react';

import icone1 from '../../img/icone-01.svg';
import icone2 from '../../img/icone-02.svg';
import icone3 from '../../img/icone-03.svg';
import icone4 from '../../img/icone-04.svg';

export default () => (
  <div className="container">
    <div className="row">
      <div className="col-sm-12 col-md-6 col-lg-3">
        <figure className="passos passo1">
          <img className="icone" src={icone1} alt="Icone do questionário" />
        </figure>
        <p className="bold">Você responde um questionário de (no mínimo) 21 questões.</p>
      </div>
      <div className="col-sm-12 col-md-6 col-lg-3">
        <figure className="passos passo2">
          <img
            className="icone"
            src={icone2}
            alt="Icone do cruzamento das suas respostas com as dos candidatos e candidatas."
          />
        </figure>
        <p>Nós cruzamos as suas respostas com os candidatos e candidatas cadastradas.</p>
      </div>
      <div className="col-sm-12 col-md-6 col-lg-3">
        <figure className="passos passo3">
          <img
            className="icone"
            src={icone3}
            alt="Icone 3 dos candidatos com mais afinidade com você."
          />
        </figure>
        <p>Te informamos quem dos candidatos e candidatas pensam mais parecido com você.</p>
      </div>
      <div className="col-sm-12 col-md-6 col-lg-3">
        <figure className="passos passo4">
          <img
            className="icone"
            src={icone4}
            alt="Icone 4. Você pode olhar as respostas de todos os candidatos e candidatas."
          />
        </figure>
        <p>Você pode olhar as respostas e perfil de todos eles e elas.</p>
      </div>
    </div>
  </div>
);