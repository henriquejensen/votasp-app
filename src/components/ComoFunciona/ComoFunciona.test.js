import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ComoFunciona from './ComoFunciona';

describe('ComoFunciona', () => {
  it('should match a snapshot', () => {
    expect(toJson(shallow(( <ComoFunciona /> )))).toMatchSnapshot();
  })
})
