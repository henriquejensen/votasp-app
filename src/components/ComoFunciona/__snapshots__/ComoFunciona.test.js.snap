// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComoFunciona should match a snapshot 1`] = `
<PageLayout>
  <section
    className="como-funciona"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <h2
            className="upper text-center titulo"
          >
            Como Funciona?
          </h2>
        </div>
      </div>
      <div
        className="row descricao"
      >
        <h4
          className="text-center"
        >
          O Vota SP é um site que vai ajudar a população do Estado de São Paulo escolher seus candidatos e candidatas a deputado, deputada estadual e federal.
        </h4>
      </div>
      <Component />
    </div>
  </section>
  <hr />
  <section
    className="algoritmo"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <h2
            className="upper text-center titulo"
          >
            Como é Feito o Cálculo do Ranking?
          </h2>
        </div>
        <div>
          <div
            className="row"
          >
            <div
              className="col-12"
            >
              <p
                className="bold"
              >
                1. Coletamos as respostas do eleitor ou da eleitora. 
              </p>
              <p>
                2. Para cada pergunta respondida seguimos os seguintes passos:
              </p>
              <div
                className="passosItem2"
              >
                <p>
                  a. Comparamos a resposta do eleitor ou da eleitora com a resposta do candidato ou candidata e pontuamos de acordo com a tabela abaixo:
                </p>
                <div>
                  <table
                    className="tabelaRanking"
                  >
                    <thead
                      className="bold"
                    >
                      <tr>
                        <td>
                          Resposta do Eleitor(a)
                        </td>
                        <td>
                          Resposta do Candidato(a)
                        </td>
                        <td>
                          Pontuação do Candidato(a)
                        </td>
                      </tr>
                    </thead>
                    <tbody>
                      <tr>
                        <td>
                          Discordo Plenamente
                        </td>
                        <td>
                          Discordo Plenamente
                        </td>
                        <td
                          className="number"
                        >
                          4
                        </td>
                      </tr>
                      <tr>
                        <td>
                          Discordo Plenamente
                        </td>
                        <td>
                          Discordo
                        </td>
                        <td
                          className="number"
                        >
                          2
                        </td>
                      </tr>
                      <tr>
                        <td>
                          Discordo Plenamente
                        </td>
                        <td>
                          Concordo
                        </td>
                        <td
                          className="number"
                        >
                          -3
                        </td>
                      </tr>
                      <tr>
                        <td>
                          Discordo Plenamente
                        </td>
                        <td>
                          Concordo Plenamente
                        </td>
                        <td
                          className="number"
                        >
                          -4
                        </td>
                      </tr>
                      <tr>
                        <td>
                          Discordo
                        </td>
                        <td>
                          Discordo Plenamente
                        </td>
                        <td
                          className="number"
                        >
                          2
                        </td>
                      </tr>
                      <tr>
                        <td>
                          Discordo
                        </td>
                        <td>
                          Discordo
                        </td>
                        <td
                          className="number"
                        >
                          4
                        </td>
                      </tr>
                      <tr>
                        <td>
                          Discordo
                        </td>
                        <td>
                          Concordo
                        </td>
                        <td
                          className="number"
                        >
                          -2
                        </td>
                      </tr>
                      <tr>
                        <td>
                          Discordo
                        </td>
                        <td>
                          Concordo Plenamente
                        </td>
                        <td
                          className="number"
                        >
                          -3
                        </td>
                      </tr>
                      <tr>
                        <td>
                          Concordo
                        </td>
                        <td>
                          Discordo Plenamente
                        </td>
                        <td
                          className="number"
                        >
                          -3
                        </td>
                      </tr>
                      <tr>
                        <td>
                          Concordo
                        </td>
                        <td>
                          Discordo
                        </td>
                        <td
                          className="number"
                        >
                          -2
                        </td>
                      </tr>
                      <tr>
                        <td>
                          Concordo
                        </td>
                        <td>
                          Concordo
                        </td>
                        <td
                          className="number"
                        >
                          4
                        </td>
                      </tr>
                      <tr>
                        <td>
                          Concordo
                        </td>
                        <td>
                          Concordo Plenamente
                        </td>
                        <td
                          className="number"
                        >
                          2
                        </td>
                      </tr>
                      <tr>
                        <td>
                          Concordo Plenamente
                        </td>
                        <td>
                          Discordo Plenamente
                        </td>
                        <td
                          className="number"
                        >
                          -4
                        </td>
                      </tr>
                      <tr>
                        <td>
                          Concordo Plenamente
                        </td>
                        <td>
                          Discordo
                        </td>
                        <td
                          className="number"
                        >
                          -3
                        </td>
                      </tr>
                      <tr>
                        <td>
                          Concordo Plenamente
                        </td>
                        <td>
                          Concordo
                        </td>
                        <td
                          className="number"
                        >
                          2
                        </td>
                      </tr>
                      <tr>
                        <td>
                          Concordo Plenamente
                        </td>
                        <td>
                          Concordo Plenamente
                        </td>
                        <td
                          className="number"
                        >
                          4
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
                <p>
                  b. Somamos todos os pontos obtidos e normalizamos os dados para obter o percentual de afinidade.
                   
                </p>
                <div
                  className="passo2Itemb"
                >
                  <p>
                    i. O máximo de pontos que um candidato ou candidata pode obter é o número de respostas do eleitor ou eleitora multiplicado por 4.
                  </p>
                  <p>
                    ii. Suponha que o eleitor ou eleitora tenha respondido 20 questões. O máximo portanto seria 20 * 4 = 80 pontos.
                     
                  </p>
                  <p>
                    iii. Se o candidato ou candidata, a partir da tabela acima, obteve 60 pontos, dividimos 60 por 80, o que resulta no valor 0,75.
                  </p>
                  <p>
                    iv. Em seguida, multiplicamos esse valor por 100, o que resulta em um índice de afinidade de 75%, considerando assim as 20 respostas que o eleitor ou a eleitora respondeu.
                  </p>
                </div>
                <p>
                  c. Fazemos esse mesmo cálculo para todos os candidatos e candidatas.
                </p>
                <p>
                  d. Ordenamos de forma descendente pela pontuação de cada candidato ou candidata e exibimos uma lista por ordem de afinidade.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</PageLayout>
`;
